Editor/CKEditor/Resources/views/editorheader.html.twig

```diff
@@ -26,64 +31,135 @@
+    editorOptions.toolbarGroups = [
+		{ name: 'tools', groups: [ 'tools', 'cleanup', 'undo' ] },
+		{ name: 'basicstyles', groups: [ 'basicstyles', 'align', 'list' ] },
+		{ name: 'links', groups: [ 'links', 'image' ] },
+		{ name: 'insert', groups: [ 'media', 'insert' ] }
+	];
+{% elseif editorVars.barmode == 'Simple' %}
+    editorOptions.removeButtons += ',CopyFormatting,autoFormat,SelectAll,Strike,Subscript,Superscript,Outdent,Indent,Blockquote,CreateDiv,Unlink,Anchor,Flash,HorizontalRule,Smiley,SpecialChar,PageBreak,Iframe,btgrid,ckawesome';
+    editorOptions.removeButtons += ',searchCode,CommentSelectedRange,UncommentSelectedRange,AutoComplete';
+    editorOptions.toolbarGroups = [
+		{ name: 'tools', groups: [ 'tools', 'clipboard', 'cleanup', 'undo' ] },
+		{ name: 'basicstyles', groups: [ 'basicstyles' ] },
+		{ name: 'paragraph', groups: [ 'align', 'list', 'indent', 'bidi' ] },
+		{ name: 'links', groups: [ 'links', 'image', 'Uploadcare' ] },
+		{ name: 'insert', groups: [ 'media', 'insert',  'others' ] },
+		{ name: 'about', groups: [ 'about' ] }
+	];
+{% elseif editorVars.barmode == 'Standard' %}
+    editorOptions.removeButtons += ',CopyFormatting,autoFormat,SelectAll,Anchor,btgrid,Print';
+    editorOptions.removeButtons += ',Styles,Format';
+    editorOptions.removeButtons += ',searchCode,CommentSelectedRange,UncommentSelectedRange,AutoComplete';
+    editorOptions.removePlugins += ',zoom,videosnapshot,lineheight';
+    editorOptions.toolbarGroups = [
+		{ name: 'tools', groups: [ 'tools', 'mode', 'clipboard', 'cleanup', 'undo' ] },
+		{ name: 'editing', groups: [ 'find', 'selection', 'spellchecker' ] },
+		{ name: 'links', groups: [ 'links', 'image', 'Uploadcare', 'media', 'insert' ] },
+		'/',
+		{ name: 'basicstyles', groups: [ 'basicstyles', 'texttransform' ] },
+		{ name: 'paragraph', groups: [ 'align', 'list', 'indent', 'blocks', 'bidi' ] },
+		{ name: 'styles', groups: [ 'styles', 'colors' ] },
+		{ name: 'others', groups: [ 'others' ] },
+		{ name: 'about', groups: [ 'about' ] }
+	];
+{% else %}
+    editorOptions.extraPlugins += ',codeTag';
+    editorOptions.toolbarGroups = [
+		{ name: 'tools', groups: [ 'tools' ] },
+		{ name: 'mode', groups: [ 'mode' ] },
+		{ name: 'document', groups: [ 'document', 'doctools', 'clipboard', 'cleanup', 'undo' ] },
+		{ name: 'editing', groups: [ 'find', 'selection', 'spellchecker', 'editing' ] },
+		{ name: 'forms', groups: [ 'forms' ] },
+		'/',
+		{ name: 'basicstyles', groups: [ 'basicstyles' ] },
+		{ name: 'paragraph', groups: [ 'align', 'list', 'indent', 'paragraph' ] },
+		{ name: 'links', groups: [ 'links', 'image', 'Uploadcare' ] },
+		{ name: 'insert', groups: [ 'media', 'insert' ] },
+		'/',
+		{ name: 'texttransform', groups: [ 'texttransform' ] },
+		{ name: 'styles', groups: [ 'blocks', 'bidi', 'styles' ] },
+		{ name: 'colors', groups: [ 'colors' ] },
+		{ name: 'others', groups: [ 'others' ] },
+		{ name: 'about', groups: [ 'about' ] }
+	];
+{% endif %}
```

Comment:
Why are you hardcoding all toolbar layouts in the template? Wasn't there a custconfig file for these definitions? Or did you consider this for easier customisation?

Reply:
I thought long before I decided how to implement toolbars, and to have comparability. I will try to explain.
In CKEditor they have too approaches to configure toolbar: Group Configuration, and “Item by Item” Configuration. Here https://docs.ckeditor.com/ckeditor4/latest/guide/dev_toolbarconcepts.html 
two concepts are described in details.

Proffered is group configuration. From above link: The most important benefit of toolbar grouping configuration over the “item by item” configuration is: automation. It is now possible for plugin developers to define into which toolbar group their plugins should add buttons. For example, the Image plugin includes its button in the insert group, while the Undo and Redo buttons go into the undo sub-group. .... ... ...

To the moment our overwrite system uses “Item by Item” Configuration and this is OK to select concrete buttons for the toolbar. Now Basic, Simple and Standard toolbars in our Scribite CKEditor settings use toolbar configration in custconfig.js.

Extended (Full) setting uses group configuration with all benefits.

I agree group config is better to move to custcongig.js, but I have had difficulty defining them there.

I am sure we will find right place.